src/nvim/api/win_config.c

```diff
@@ -730,7 +730,16 @@ static bool parse_float_config(Dict(float_config) *config, FloatConfig *fconfig,
   }
 
   if (relative_is_win) {
-    fconfig->window = curwin->handle;
+    if (reconf) {
+      win_T *wp = handle_get_window(config->win);
```

Comment:
I may be wrong but couldn't you just check that `config->win != curwin->handle` instead of fetching and comparing the pointers? That way this could still be a single statement like below and no checks for the actual window need to be done in `parse_float_config`, since validation is done in `nvim_open_win` and `nvim_win_set_config`.

```lua
if (!reconf || config->win != curwin->handle) {
  fconfig->window = curwin=>handle;
}
```

Reply:
You may also need to account for `config->win = 0` (I don't think `handle_get_window` does this?)

Reply:
emm i guess we can do it early .